src/telemetry_subscriber.rs

```diff
@@ -201,24 +217,29 @@ impl Subscriber for TelemetrySubscriber {
     }
 
     fn clone_span(&self, id: &Id) -> Id {
-        if let Some(mut span_data) = self.spans.get_mut(id) {
-            // should always be present
+        if let Some(rw_lock) = self.spans.get(id_to_idx(id)) {
+            let mut span_data = rw_lock.write().unwrap();
```

Comment:
seems pretty heavyweight to me to have to acquire a write lock to just increment an integer, esp. when (on x86) all integer operations are already atomic. if the refcount was an `AtomicUsize` this could only need a read lock, or ideally, it could be moved out of the lock entirely. Something like
```rust
RefCnt<RwLock<SpanData>>
```
where the RefCt type is changed to
```rust
struct RefCt<T> {
    inner: T,
    ref_ct: AtomicUsize,
}
```